lib/howl/config.moon

```diff
@@ -101,36 +102,57 @@ define = (var = {}) ->
   defs[var.name] = var
   broadcast var.name, var.default, false
 
-set = (name, value) ->
+define_layer = (name, def={}) ->
+  layer_defs[name] = moon.copy(def)
+
+set = (name, value, scope='', layer='default') ->
   def = get_def name
 
-  if def.scope and def.scope == 'local'
-    error 'Attempt to set a global value for local variable "' .. name .. '"', 2
+  error "Unknown layer '#{layer}'" if not layer_defs[layer]
+
+  if def.scope
+    if def.scope == 'local' and scope == ''
+      error 'Attempt to set a global value for local variable "' .. name .. '"', 2
+    if def.scope == 'global' and scope != ''
+     error 'Attempt to set a local value for global variable "' .. name .. '"', 2
 
   value = convert def, value
   validate def, value
 
-  values[name] = value
-  broadcast name, value, false
+  unless scopes[scope]
+    scopes[scope] = {}
+  unless scopes[scope][name]
+    scopes[scope][name] = {}
 
-proxy_set = (name, value, proxy) ->
-  def = get_def name
+  scopes[scope][name][layer] = value
 
-  if def.scope and def.scope == 'global'
-    error 'Attempt to set a local value for global variable "' .. name .. '"', 2
+  broadcast name, value, (scope != '' or layer != 'default')
```

Comment:
I'm not sure if that would be useful. If anything, `watch` could potentially take a scope and/or layer but I think we should add that only if needed.